### ISSUE TITLE
bug: Websocket transport fails systematically after 1.9.5 version

### ISSUE THREAD

**Issue by korbav** [wagmi-dev/viem#1226]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.10.14

### Current Behavior

Any version greater than 1.9.5 seems to be concerned.

```
const transport = webSocket(wssInfura, {
            timeout: 30000,
            retryCount: 5,
            retryDelay: 1000
        });
    
        const wsClient = createPublicClient({
            chain: polygonMumbai,
            transport,
        })

        wsClient.watchContractEvent({
            abi,
            address,
            onError: (e) => console.log(e),
            onLogs: (logs) => {
                console.log(logs);
            }
        });
```

When running this snippet, an exception is thrown immediately :

```
TypeError: Cannot read properties of undefined (reading 'constructor')
    at fn (C:\Users\borea\Documents\VSCodeProjects\viem-test-app-backend\node_modules\viem\src\utils\rpc.ts:160:22)
    at Timeout.exec [as _onTimeout] (C:\Users\borea\Documents\VSCodeProjects\viem-test-app-backend\node_modules\viem\src\utils\promise\createBatchScheduler.ts:54:5)
    at listOnTimeout (node:internal/timers:569:17)
    at processTimers (node:internal/timers:512:7)
```

Which refers to https://github.com/wagmi-dev/viem/blob/cf4006f799bdccfbf5765927a9fde0c20cbd1890/src/utils/rpc.ts#L161

### Expected Behavior

As 1.9.5 version and earlier one, it should connect properly without throwing an exception.

### Steps To Reproduce

Just run the following snippet: 

```
const transport = webSocket(wssInfura, {
            timeout: 30000,
            retryCount: 5,
            retryDelay: 1000
        });
    
        const wsClient = createPublicClient({
            chain: polygonMumbai,
            transport,
        })

        wsClient.watchContractEvent({
            abi,
            address,
            onError: (e) => console.log(e),
            onLogs: (logs) => {
                console.log(logs);
            }
        });
```

### Anything else?

It might be related to this closed issue https://github.com/wagmi-dev/viem/issues/1174

**Comment by jxom** [comment]
Can you please provide a minimal reproducible example? It is hard to tell what is going on without knowing your environment. Are you running this in Node, Deno, Webpack, Vite, Rollup, Bun, etc? What version?

**Comment by jxom** [comment]
Closing until a reproduction is provided.

**Comment by korbav** [comment]
> Closing until a reproduction is provided.

Here you go: [https://codesandbox.io/p/sandbox/gifted-chaplygin-xt675d?welcome=true](https://codesandbox.io/p/sandbox/gifted-chaplygin-xt675d?welcome=true)



`npm run start` on the bottom terminal to launch it
Note: Version 1.11.1 is used, but same goes with any version after 1.9.5

**Comment by jxom** [comment]
Going to integrate our own isomorphic `ws` implementation this week: https://github.com/wagmi-dev/isows and hopefully resolve this issue once and for all!

**Comment by jxom** [comment]
Should be fixed in `1.15.5`!
